pkg/pool-utils/contracts/oracle/PoolPriceOracle.sol

```diff
@@ -77,6 +77,24 @@ abstract contract PoolPriceOracle is IPoolPriceOracle, IPriceOracle {
         return Buffer.SIZE;
     }
 
+    /**
+     * @dev Manually fill oracle slots with dummy data, so that subsequent swaps filling them with real data
+     * cost less gas. Initialize by range, since filling a slot costs roughly 20k gas, so trying to fill the
+     * entire buffer at once would run out of gas.
+     */
+    function initializeOracle(uint256 startSlot, uint256 endSlot) external {
+        _require(startSlot < endSlot && endSlot <= Buffer.SIZE, Errors.OUT_OF_BOUNDS);
+
+        // Cannot use zero, or it will not actually do anything (uninitialized entries are already 0)
+        // Cannot use 0x01, since the timestamp is in the LSB position, and needs to be zero for invalid entries
+        bytes32 initSample = bytes32(0xba10000000000000000000000000000000000000000000000000000000000000);
+        for (uint256 i = startSlot; i < endSlot; i++) {
+            if (_samples[i] == 0) {
```

Comment:
```suggestion
            if (_samples[i].timestamp() == 0) {
```

Reply:
I had it like that so it didn't overwrite slots it already initialized. If someone initialized 100-200, and then somebody else did 150-250, the second one would only write to 201-250. I suppose it doesn't hurt (or cost as much gas) to overwrite something. Is it cheaper to check timestamp() == 0 vs the whole thing 0?

Reply:
Sorry, that's not the point of my change. The point is that we're currently relying on 'timestamp != 0' to mean 'slot is uninitialized' - we should use that throughout (maybe even include a method that returns a bool).

This doesn't change the behavior of the contract.